homeassistant/helpers/entity_registry.py

```diff
@@ -841,3 +842,25 @@ async def async_migrate_entries(
 
         if updates is not None:
             ent_reg.async_update_entity(entry.entity_id, **updates)
+
+
+def async_resolve_entity_ids(
+    registry: EntityRegistry, entity_ids_or_uuids: list[str]
+) -> list[str]:
+    """Resolve a list of entity ids or UUIDs to a list of entity ids."""
+
+    def resolve_entity(entity_id_or_uuid: str) -> str | None:
+        """Resolve an entity id or UUID to an entity id or None."""
+        if valid_entity_id(entity_id_or_uuid):
+            return entity_id_or_uuid
+        for entry in registry.entities.values():
```

Comment:
This is too inefficient, especially on large installations. Let's keep an index of uuid -> entry.

Reply:
Make sure to add tests that adding/removing keeps the index up to date.

Reply:
Fixed in 1a05c1b